pkg/providers/ovirt/validation/vm-import-validation.go

```diff
@@ -0,0 +1,124 @@
+	validators.DiskBackupID:                        warn,
+	validators.DiskLunStorageID:                    block,
+	validators.DiskPropagateErrorsID:               log,
+	validators.DiskWipeAfterDeleteID:               log,
+	validators.DiskStatusID:                        block,
+	validators.DiskStoragaTypeID:                   block,
+	validators.DiskSgioID:                          block,
+	// VM rules
+	validators.VMBiosBootMenuID:                  log,
+	validators.VMBiosTypeID:                      block,
+	validators.VMBiosTypeQ35SecureBootID:         warn,
+	validators.VMCpuArchitectureID:               block,
+	validators.VMCpuTuneID:                       warn,
+	validators.VMCpuSharesID:                     log,
+	validators.VMCustomPropertiesID:              warn,
+	validators.VMDisplayTypeID:                   log,
+	validators.VMHasIllegalImagesID:              block,
+	validators.VMHighAvailabilityPriorityID:      log,
+	validators.VMIoThreadsID:                     warn,
+	validators.VMMemoryPolicyBallooningID:        log,
+	validators.VMMemoryPolicyOvercommitPercentID: log,
+	validators.VMMemoryPolicyGuaranteedID:        log,
+	validators.VMMigrationID:                     log,
+	validators.VMMigrationDowntimeID:             log,
+	validators.VMNumaTuneModeID:                  warn,
+	validators.VMOriginID:                        block,
+	validators.VMRngDeviceSourceID:               log,
+	validators.VMSoundcardEnabledID:              warn,
+	validators.VMStartPausedID:                   log,
+	validators.VMStorageErrorResumeBehaviourID:   log,
+	validators.VMTunnelMigrationID:               warn,
+	validators.VMUsbID:                           block,
+	validators.VMGraphicConsolesID:               log,
+	validators.VMHostDevicesID:                   log,
+	validators.VMReportedDevicesID:               log,
+	validators.VMQuotaID:                         log,
+	validators.VMWatchdogsID:                     block,
+	validators.VMCdromsID:                        log,
+	validators.VMFloppiesID:                      log,
+}
+
+// Validator validates different properties of a VM
+type Validator interface {
+	ValidateVM(vm *ovirtsdk.Vm) []validators.ValidationFailure
+	ValidateDiskAttachments(diskAttachments []*ovirtsdk.DiskAttachment) []validators.ValidationFailure
+	ValidateNics(nics []*ovirtsdk.Nic) []validators.ValidationFailure
+}
+
+// VirtualMachineImportValidator validates VirtualMachineImport object
+type VirtualMachineImportValidator struct {
+	Validator Validator
+}
+
+// NewVirtualMachineImportValidator creates ready-to-use NewVirtualMachineImportValidator
+func NewVirtualMachineImportValidator() VirtualMachineImportValidator {
+	return VirtualMachineImportValidator{
+		Validator: &validators.ValidatorWrapper{},
+	}
+}
+
+// Validate validates whether VM described in VirtualMachineImport can be imported
+func (validator *VirtualMachineImportValidator) Validate(vm *ovirtsdk.Vm) ([]validators.ValidationFailure, bool) {
```

Comment:
Where do we need `[]ValidationFailure` slice out side of this method?

Reply:
In code that will add the failure information to the CR - probably in the reconciliation loop.

Reply:
It seems like `processFailures` is not really processing `ValidationFailure`s. We log and filter log action and return everything else.

Reply:
You're right. To make it really process all the failures I will change the API  from `Validate(*ovirtsdk.vm)` to `Validate(v2vvalpha1.VirtualMachineImport)` and update the CR status in that function for warns and blocks.

Reply:
Just pushed the changes to enclose whole validation processing in that package. Please have a look if that's better.